state/raftlease/target.go

```diff
@@ -247,13 +326,24 @@ func MakeTrapdoorFunc(mongo Mongo, collection string) raftlease.TrapdoorFunc {
 
 func getRecord(coll mongo.Collection, docId string) (leaseHolderDoc, error) {
 	var doc leaseHolderDoc
-	err := coll.FindId(docId).One(&doc)
-	if err != nil {
-		return leaseHolderDoc{}, err
+	if err := coll.FindId(docId).One(&doc); err != nil {
+		return leaseHolderDoc{}, errors.Trace(err)
 	}
 	return doc, nil
 }
 
+func getRecords(coll mongo.Collection, docIds []string) ([]leaseHolderDoc, error) {
+	var docs []leaseHolderDoc
+	if err := coll.Find(bson.M{
+		"_id": bson.M{
+			"$in": docIds,
+		},
+	}).Sort("_id").All(&docs); err != nil {
```

Comment:
I don't think we care about Sorted order, do we? We iterate over them and lookup in a map. I don't feel terrible if we sort, but it is extra processing that we wouldn't have to do.

Reply:
I wanted it to be deterministic when creating the `txn.Op`.